library/core/src/iter/adapters/step_by.rs

```diff
@@ -233,3 +225,96 @@ where
 // StepBy can only make the iterator shorter, so the len will still fit.
 #[stable(feature = "iterator_step_by", since = "1.28.0")]
 impl<I> ExactSizeIterator for StepBy<I> where I: ExactSizeIterator {}
+
+trait SpecRange: Iterator {
+    fn spec_fold<Acc, F>(self, acc: Acc, f: F) -> Acc
+    where
+        F: FnMut(Acc, Self::Item) -> Acc;
+}
+
+impl<I> SpecRange for StepBy<I>
+where
+    I: Iterator,
+{
+    default fn spec_fold<Acc, F>(mut self, mut acc: Acc, mut f: F) -> Acc
+    where
+        F: FnMut(Acc, Self::Item) -> Acc,
+    {
+        #[inline]
+        fn nth<I: Iterator>(iter: &mut I, step: usize) -> impl FnMut() -> Option<I::Item> + '_ {
+            move || iter.nth(step)
+        }
+
+        if self.first_take {
+            self.first_take = false;
+            match self.iter.next() {
+                None => return acc,
+                Some(x) => acc = f(acc, x),
+            }
+        }
+        from_fn(nth(&mut self.iter, self.step)).fold(acc, f)
+    }
+}
+
+macro_rules! spec_int_ranges {
+    ($($t:ty)*) => ($(
+        impl SpecRange for StepBy<Range<$t>> {
+            #[inline]
+            fn spec_fold<Acc, F>(self, mut acc: Acc, mut f: F) -> Acc
+            where
+                F: FnMut(Acc, Self::Item) -> Acc
+            {
+                use crate::convert::TryFrom;
```

Comment:
Isn't `core` in 2021?  Why is this needed?

Reply:
```
error[E0599]: no function or associated item named `try_from` found for type `usize` in the current scope
   --> library/core/src/iter/adapters/step_by.rs:290:39
    |
260 | / macro_rules! spec_int_ranges {
261 | |     ($($t:ty)*) => ($(
262 | |         impl SpecRange for StepBy<Range<$t>> {
263 | |             #[inline]
...   |
290 | |                 let inner_len = <$t>::try_from(range.size_hint().0).unwrap_or(<$t>::MAX);
    | |                                       ^^^^^^^^ function or associated item not found in `usize`
...   |
311 | |     )*)
312 | | }
    | |_- in this expansion of `spec_int_ranges!`
...
315 |   spec_int_ranges!(u8 u16 u32 u64 usize);
    |   -------------------------------------- in this macro invocation
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
1   + use crate::convert::TryFrom;
    |
```

Maybe due to macros?